### DISCUSSION TITLE
New user

### DISCUSSION THREAD

**Discussion by agregorynz** [monocasual/giada#623] [Ideas]
Gilda looks really promising as a live performance environment. Looking forward to explore deeper. I’m curious to know if there are plans to be able to make or introduce these kind of features.
1. Play and sequence sample action list with a piano roll so sample patterns can be played chromatically instead of just one shot ‘hits’
2. Sample off-set commands like in a tracker to start pattern from within sample. 
3. Or a beat slicer function to be used with the piano roll.
4. Sample slot to include envelope and looping controls (forward loop, ping pong, x fade control). 
5. VST2 support would be amazing considering the huge amount of plugins available.
6. Scene buttons.
7. Change clip colours.

Very cool so far. Keep chipping away at developing this fantastic program.

**Comment by gvnnz** [comment]
Hi @agregorynz, welcome and thank you for your kind words. We have some cool plans for the future regarding the development of Giada. Going over your list:

> 6. Scene buttons.

Absolutely. We call it Sequencer Banks (but "Scenes" sounds better, actually) https://github.com/monocasual/giada/issues/64. Will be added after 1.0 release

> 7. Change clip colours.

That's another useful feature planned for version 1.x: https://github.com/monocasual/giada/issues/312

> 2. Sample off-set commands like in a tracker to start pattern from within sample.

You can already do that via the "offset" value in the Sample Editor, if I get your point right. 

> 1. Play and sequence sample action list with a piano roll so sample patterns can be played chromatically instead of just one shot ‘hits’
> 3. Or a beat slicer function to be used with the piano roll.
> 4. Sample slot to include envelope and looping controls (forward loop, ping pong, x fade control).

We thought a lot about these. We will probably add no.1, the ability to play samples chromatically (maybe also via a virtual keyboard, aka the computer keyboard). On the other hand there are many cool VST samplers out there, even open source ones, that you can drop in a MIDI channel and do crazy stuff with them. Adding those features to Giada feels like reinventing the wheel a little bit. We'll see...

> 5. VST2 support would be amazing considering the huge amount of plugins available.

Unfortunately VST2 has been deprecated by Steinberg and VST2 support cannot be shipped anymore. If you feel brave enough you can always build Giada yourself by including the right VST2.x SDK, there's also an unofficial one around the Internet if I remember correctly. More info on how to compile Giada [here](https://www.giadamusic.com/documentation-compiling-from-source).

That said, any kind of feedback is very welcome around here. Feel free to add yours anytime ;)

**Reply by agregorynz** [reply]
Hi Giovani,
Thanks for the reply, great to see that there are some great devopment plans in the works. 
To explain my offset question a little further, in a tracker, a musician is able to play a note and program the start position within the sample as it starts to play. For example, if you had a vocal phrase that you wanted to play sequence a stutter effect or repeat a word rhythmically, you would set the offset command to set the start position for the sampler play head. It has the advantage of not having to chop up and copy a larger sample into many smaller pieces because you only need to write the additional command. 

I understand what you are saying about not wanting to reinvent the wheel in regards to samplers. I was going to try to use 'Grace', 'Poise' and 'Sitala' samplers but unfortunately they are all VST2. I am not really sure about what simple VST3 samplers there are available out there. Do you have some recccomendations? 

Regards, 

Alan

**Comment by gvnnz** [comment]
Hey @agregorynz , 

> To explain my offset question a little further, in a tracker, a musician is able to play a note and program the start position within the sample as it starts to play. For example, if you had a vocal phrase that you wanted to play sequence a stutter effect or repeat a word rhythmically, you would set the offset command to set the start position for the sampler play head. It has the advantage of not having to chop up and copy a larger sample into many smaller pieces because you only need to write the additional command.

Thanks, I see now. Yes, currently the only way is to chop up the sample and make a new channel for each chunk of audio you need to play. However there's the "to new channel" function in the Sample Editor that should alleviate the pain of doing it by hand.

>  I am not really sure about what simple VST3 samplers there are available out there. Do you have some recccomendations?

I keep an eye on [Bliss by DiscoDSP](https://www.discodsp.com/bliss/), not free but very cool. 
Or the whole open-source [LSP plug-ins suite](https://lsp-plug.in/), (LV2 format) which features a couple of interesting samplers.
